pkg/requester/ranking/previous_executions.go

```diff
@@ -47,19 +47,25 @@ func (s *PreviousExecutionsNodeRanker) RankNodes(ctx context.Context, job model.
 	}
 	for i, node := range nodes {
 		rank := 3 * requester.RankPreferred
+		reason := "job not executed yet"
 		if previousExecutions, ok := previousExecutors[node.PeerInfo.ID.String()]; ok {
 			if previousExecutions > 1 {
 				rank = requester.RankUnsuitable
+				reason = "job already executed on this node"
 			} else if _, filterOut := toFilterOut[node.PeerInfo.ID.String()]; filterOut {
 				rank = requester.RankUnsuitable
+				reason = "job rejected or produced invalid result"
 			} else {
 				rank = requester.RankPossible
+				reason = "job already executed on this node"
```

Comment:
This is slightly confusing as it matches the reason on line 54, but with a different rank.  Is it possible to clarify the else block is for `previousExecutions == 1` and with a slightly different reason?